test/local_pow_test.py

```diff
@@ -0,0 +1,197 @@
+            # Result is the one returned by MockAdapter
+            self.assertEqual(result['trytes'], self.bundle)
+            # And not by mocked pow pkg
+            self.assertNotEqual(result['trytes'], self.ccurl_bundle)
+
+    def test_http_adapter(self):
+        """
+        Test if local_pow feature works with HttpAdapter.
+        """
+        # Note that we need correct return value to pass the
+        # repsonse filter.
+        with patch('pow.ccurl_interface.attach_to_tangle',
+                    MagicMock(return_value=self.bundle)) as mocked_ccurl:
+            api = Iota(HttpAdapter('http://localhost:14265/'),local_pow=True)
+            result = api.attach_to_tangle(
+                self.trunk,
+                self.branch,
+                self.bundle,
+                self.mwm)
+            self.assertTrue(mocked_ccurl.called)
+            self.assertEqual(result['trytes'], self.bundle)
+
+    def test_mock_adapter(self):
+        """
+        Test if local_pow feature works with MockAdapter.
+        """
+        # Note that we need correct return value to pass the
+        # repsonse filter.
+        with patch('pow.ccurl_interface.attach_to_tangle',
+                    MagicMock(return_value=self.bundle)) as mocked_ccurl:
+            api = Iota(MockAdapter(),local_pow=True)
+            result = api.attach_to_tangle(
+                self.trunk,
+                self.branch,
+                self.bundle,
+                self.mwm)
+            self.assertTrue(mocked_ccurl.called)
+            self.assertEqual(result['trytes'], self.bundle)
+
+    def test_sandbox_adapter(self):
+        """
+        Test if local_pow feature works with SandboxAdapter.
+        """
+        # Note that we need correct return value to pass the
+        # repsonse filter.
+        with patch('pow.ccurl_interface.attach_to_tangle',
+                    MagicMock(return_value=self.bundle)) as mocked_ccurl:
+            api = Iota(SandboxAdapter('https://sandbox.iota:14265/api/v1/', auth_token=None),
+                       local_pow=True)
+            result = api.attach_to_tangle(
+                self.trunk,
+                self.branch,
+                self.bundle,
+                self.mwm)
+            self.assertTrue(mocked_ccurl.called)
+            self.assertEqual(result['trytes'], self.bundle)
+    
+    def test_routing_wrapper(self):
+        """
+        Test if local_pow feature works with RoutingWrapper.
+        """
+        # Note that we need correct return value to pass the
+        # repsonse filter.
+        with patch('pow.ccurl_interface.attach_to_tangle',
+                    MagicMock(return_value=self.bundle)) as mocked_ccurl:
+            # We are trying to redirect `attach_to_tangle` calls to localhost
+            # with a RoutingWrapper. However, if local_pow=true, the pow
+            # request will not reach the adapter, but will be directed to
+            # ccurl interface. 
```

Comment:
💭 Ah, very interesting; that makes sense.  Should we also modify the documentation for `RoutingWrapper` to call this out?
﻿
﻿﻿https://pyota.readthedocs.io/en/latest/adapters.html#routingwrapper

Reply:
Done ;)